presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -546,18 +546,22 @@ PrestoServer::getHttpServerFilters() {
   }
 
   if (SystemConfig::instance()->enableHttpStatsFilter()) {
-    auto filter = getHttpStatsFilter();
-    if (filter != nullptr) {
-      filters.push_back(std::move(filter));
+    auto additionalFilters = getAdditionalHttpStatsFilters();
+    if (!additionalFilters.empty()) {
```

Comment:
The empty check is not needed, since the iteration won't go forward if list is empty.